chi.go

```diff
@@ -1,81 +1,58 @@
 package chi
 
 import (
+	"context"
 	"net/http"
-
-	"golang.org/x/net/context"
 )
 
 // NewRouter returns a new Mux object that implements the Router interface.
 // It accepts an optional parent context.Context argument used by all
 // request contexts useful for signaling a server shutdown.
+
+// TODO: remove this parent context....? does it work..? do we need it?
+// or do we add our own signal to each routeContext ...?
+
 func NewRouter(parent ...context.Context) *Mux {
 	return NewMux(parent...)
 }
 
-// A Router consisting of the core routing methods used by chi's Mux.
-//
-// NOTE, the plan: hopefully once net/context makes it into the stdlib and
-// net/http supports a request context, we will remove the chi.Handler
-// interface, and the Router argument types will be http.Handler instead
-// of interface{}.
+// A Router consisting of the core routing methods used by chi's Mux,
+// built using just the stdlib.
 type Router interface {
 	http.Handler
-	Handler
-
-	Use(middlewares ...interface{})
-	Group(fn func(r Router)) Router
-	Route(pattern string, fn func(r Router)) Router
-	Mount(pattern string, handlers ...interface{})
-
-	Handle(pattern string, handlers ...interface{})
-	NotFound(h HandlerFunc)
 
-	Connect(pattern string, handlers ...interface{})
-	Head(pattern string, handlers ...interface{})
-	Get(pattern string, handlers ...interface{})
-	Post(pattern string, handlers ...interface{})
-	Put(pattern string, handlers ...interface{})
-	Patch(pattern string, handlers ...interface{})
-	Delete(pattern string, handlers ...interface{})
-	Trace(pattern string, handlers ...interface{})
-	Options(pattern string, handlers ...interface{})
+	Use(middlewares ...func(http.Handler) http.Handler)
+	Stack(fn func(r Router)) Router
+	Group(pattern string, fn func(r Router)) Router
```

Comment:
I think you're right and I'll go back to original Route() and Group() for now

Route is like a Mount that accepts a function instead of the router, I like it for building the tree as shorthand 

> On Jun 15, 2016, at 11:32 PM, Vojtech Vitek <notifications@github.com> wrote:
> 
> In chi.go:
> 
> >  
> > -	Connect(pattern string, handlers ...interface{})
> > -	Head(pattern string, handlers ...interface{})
> > -	Get(pattern string, handlers ...interface{})
> > -	Post(pattern string, handlers ...interface{})
> > -	Put(pattern string, handlers ...interface{})
> > -	Patch(pattern string, handlers ...interface{})
> > -	Delete(pattern string, handlers ...interface{})
> > -	Trace(pattern string, handlers ...interface{})
> > -	Options(pattern string, handlers ...interface{})
> > +	Use(middlewares ...func(http.Handler) http.Handler)
> > +	Stack(fn func(r Router)) Router
> > +	Group(pattern string, fn func(r Router)) Router
> btw: How about removing .Route(), changing .Mount() signature (so it's not like .Handle()) and keep .Group() only?
> 
> type Router interface {
>     http.Handler
> 
>     Group(fn func(r Router)) Router
>     Mount(pattern string, r Router)
>     Handle(pattern string, h http.Handler)
>     //...
> }
> 
> // --------
> 
> r := chi.NewRouter()
> r.Use(ForAllCtx)
> 
> r.Group(func(r chi.Router) {
>     r.Use(SpecialCtx)
>     r.Get("/", handler)
> })
> 
> r.Mount("/ping", subrouter)
> 
> r.Mount("/:articleID", r.Group(func(r chi.Router) {
>     r.Use(ArticleCtx)
>     r.Get("/", getArticle)
> }))
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>